ocaml/xapi/message_forwarding.ml

```diff
@@ -1639,6 +1639,16 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 					~host_from:(Helpers.LocalObject source_host)
 					~host_to:(Helpers.LocalObject host);
 			end;
+			
+			let vm_virt_hw_vn = Db.VM.get_virt_hw_vn ~__context ~self:vm in
```

Comment:
This prevents migration but will not prevent a halted VM being started on an unsuitable host.
I think if we do this check in assert_can_boot_here instead, then it will prevent both migration and starting.